vars/neoDeploy.groovy

```diff
@@ -72,6 +72,14 @@ import static com.sap.piper.Prerequisites.checkScript
 ]
 
 @Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /**
+     * The deployment mode which should be used. Available options are:
+     * *`'mta'` - default,
+     * *`'warParams'` - deploying WAR file and passing all the deployment parameters via the function call,
+     * *`'warPropertiesFile'` - deploying WAR file and putting all the deployment parameters in a .properties file.
+     * @possibleValues 'mta', 'warParams', 'warPropertiesFile'
+     */
+    'deployMode',
```

Comment:
in order to use `neoDeploy ` with the deploy modes 'warParams' and 'warPropertiesFile', we need to move this parameter to the step config. Another option would be to duplicate it on the `multicloudDeploy `step. I can duplicate it if you would like.